### PR TITLE
LPD-37458 Paginator is not showing intermediate pages in mobile render

### PR DESCRIPTION
**Pull request by chestofwonders**
Issue: https://liferay.atlassian.net/browse/LPD-37458

<h1>Paginator is not showing intermediate pages in mobile render</h1>

The goal of this PR is to fix an issue in paginator in search container that doesn't allow it to load pages on scrolling the pages list:

image

The fix removes a Clay style that sets pages dropdown `max-height` to `none`, making impossible for the browser to fire the `scroll` event.

To make the fix work, these modules must be deployed (in this order):

```
modules/apps/frontend-theme/frontend-theme-styled
modules/apps/frontend-theme/frontend-theme-classic
```

And then, deploy:

`modules/apps/frontend-theme`

After mixed, the dropdown must show the whole list of pages:

image

### CODE REVIEW

**Review by markocikos — APPROVED** [review]
@chestofwonders LGTM! :sparkles:

**Comment by markocikos** [comment]
@chestofwonders Can we push this?

**Comment by chestofwonders** [comment]
> @chestofwonders Can we forward this, or does it need another review after force push from 5 days ago?

Yes, we can forward now. We were waiting for customer to be sure that the fix is what they needed.

**Comment by chestofwonders** [comment]
ci:forward

**Comment by chestofwonders** [comment]
ci:forward

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/search_paginator.jsp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<clay:container-fluid>
+	<liferay-ui:search-paginator
+		markupView="lexicon"
+		searchContainer="<%= searchPaginatorDisplayContext.getSearchContainer() %>"
+	/>
+</clay:container-fluid>
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/search_paginator.jsp:15]: @dsanz I see what you mean, this is not right. 

Do we maybe want a sample module for all tags? Something like `frontend-taglib/frontend-taglib-sample-web`. In it, we could have several pages:
- Clay
- Liferay UI
- Liferay Frontend
- AUI
- etc

Within these, we would need to fit a lot of tags. These could be one subpage per tag, maybe accessed with vertical navigation.

In PW, they can all be under `frontend-taglib` project, further split into `/tests/clay`, `/tests/liferay-ui,` etc.

What do you think?

Comment (dsanz) [reply modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/search_paginator.jsp:15]: The pw side is more manageable as we can mimic the exact path. I'm struggling more with the sample-web module in DXP. 

`frontend-taglib/frontend-taglib-sample-web` sounds reasonable to me, as a sort of "bag" to include any portal web tag lib sample

Then in tests we would use that same folder as you suggest, with a separate forlder per tag namespace, except for clay, which has its own sample and so it can enjoy its own folder inside pw/tests

Does that sound right to you?

Comment (markocikos) [reply modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/search_paginator.jsp:15]: Yeah, sounds good. I included Clay for completion sake, as I don't see any reason for it not to be in the bag as well. But migrating it can be a very low priority.

**Comment by chestofwonders** [comment]
ci:test:sf

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/index.js
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/index.js:1]: All new frontend code should be in Typescript.

File: modules/test/playwright/tests/frontend-taglib/frontend-taglib-sample-web/liferay-ui/searchPaginator.spec.ts
Comment (markocikos) [inline modules/test/playwright/tests/frontend-taglib/frontend-taglib-sample-web/liferay-ui/searchPaginator.spec.ts:1]: Just like there is a `liferay-ui` grouping here, we could have the same grouping in sample-web.

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/view.jsp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<%
+List<TabsItem> tabsItems = sampleDisplayContext.getTabsItems();
+%>
+
+<clay:tabs
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/view.jsp:14]: We should not use `<clay:tabs>` here, as they will result in frontend-only navigation. Instead, please use `<clay:navigation-bar>`.

There should be two levels of navigation, one matching taglib library, and the other matching individual tags.

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/ClaySampleMultiselectPropsTransformer.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+import ClayDropDown from '@clayui/drop-down';
+import ClayIcon from '@clayui/icon';
+import ClaySticker from '@clayui/sticker';
+import {getSpritemap} from '@liferay/frontend-icons-web';
+import React from 'react';
+
+const spritemap = getSpritemap();
+
+const MenuCustom = ({
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/ClaySampleMultiselectPropsTransformer.js:14]: What's the purpose of all the code in `/js` dir? Isn't just `<liferay-ui:search-paginator>` in JSP enough for this task?

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/ClaySampleMultiselectPropsTransformer.js
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/ClaySampleMultiselectPropsTransformer.js:1]: Why are all the files prefixed with `ClaySample`? `Clay` does not seem right, and `Sample` seems unnecessary.

This file in particular does not seem right, as `Multiselect` is not right for the `CustomMenu` inside. I assume this is very much WIP.

**Review by markocikos — CHANGES_REQUESTED** [review]
@chestofwonders This looks like very much WIP, but I still wanted to leave few general comments on larger directions of implementation.

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/constants/ClaySamplePortletKeys.java [hunk 4]
```diff
@@ -27,6 +27,9 @@ public class ClaySamplePortletKeys {
 	public static final String NAVIGATION_BARS_DISPLAY_CONTEXT =
 		"NAVIGATION_BARS_DISPLAY_CONTEXT";
 
+	public static final String SEARCH_PAGINATOR_DISPLAY_CONTEXT =
+		"SEARCH_PAGINATOR_DISPLAY_CONTEXT";
+
```
Comment (dsanz) [inline modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/constants/ClaySamplePortletKeys.java:32]: These bits should be removed from here as search paginator will be sampled in the other module

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/display/context/ClaySampleDisplayContext.java [hunk 5]
```diff
@@ -111,6 +111,11 @@ public List<TabsItem> getTabsItems() {
 				tabsItem.setLabel("Vertical Nav");
 				tabsItem.setPanelId("vertical_nav");
 			}
+		).add(
+			tabsItem -> {
+				tabsItem.setLabel("Search Paginator");
+				tabsItem.setPanelId("search_paginator");
+			}
```
Comment (dsanz) [inline modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/display/context/ClaySampleDisplayContext.java:118]: Similarly here, this code does not belong to this sample

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/display/context/SearchPaginatorDisplayContext.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
+/**
```
Comment (dsanz) [inline modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/display/context/SearchPaginatorDisplayContext.java:1]: This one has to be moved to the other sample as well

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/portlet/ClaySamplePortlet.java [hunk 7]
```diff
@@ -70,6 +73,10 @@ public void doDispatch(
 		renderRequest.setAttribute(
 			ClaySamplePortletKeys.NAVIGATION_BARS_DISPLAY_CONTEXT,
 			new NavigationBarsDisplayContext());
+		renderRequest.setAttribute(
```
Comment (dsanz) [inline modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/portlet/ClaySamplePortlet.java:76]: same here, the `frontend-taglib-clay-sample-web` should not receive any modification in this PR

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/init.jsp [hunk 8]
```diff
@@ -48,6 +50,7 @@ ClaySampleDisplayContext claySampleDisplayContext = (ClaySampleDisplayContext)re
 DropdownsDisplayContext dropdownsDisplayContext = (DropdownsDisplayContext)request.getAttribute(ClaySamplePortletKeys.DROPDOWNS_DISPLAY_CONTEXT);
 MultiselectDisplayContext multiselectDisplayContext = (MultiselectDisplayContext)request.getAttribute(ClaySamplePortletKeys.MULTISELECT_DISPLAY_CONTEXT);
 NavigationBarsDisplayContext navigationBarsDisplayContext = (NavigationBarsDisplayContext)request.getAttribute(ClaySamplePortletKeys.NAVIGATION_BARS_DISPLAY_CONTEXT);
+SearchPaginatorDisplayContext searchPaginatorDisplayContext = (SearchPaginatorDisplayContext)request.getAttribute(ClaySamplePortletKeys.SEARCH_PAGINATOR_DISPLAY_CONTEXT);
```
Comment (dsanz) [inline modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/init.jsp:53]: same idea, remove

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/.lfrbuild-portal
[the remark this answers is not in the document]
Comment (dsanz) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/.lfrbuild-portal:1]: curiously, `frontend-taglib-clay-sample-web` also has this marker file. We probably need to delete both:
image

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/java/com/liferay/frontend/taglib/sample/web/internal/display/context/SearchPaginatorDisplayContext.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+		SearchContainer<DropdownItem> searchContainer = new SearchContainer<>(
+			_renderRequest,
+			PortletURLUtil.getCurrent(
+				liferayPortletRequest, liferayPortletResponse),
+			null, "no-items-were-found");
+
+		searchContainer.setId("DropdownItemSearchContainer");
+		searchContainer.setOrderByCol(
+			SearchOrderByUtil.getOrderByCol(
+				_renderRequest, SamplePortletKeys.CLAY_SAMPLE, "order-by-col",
```
Comment (dsanz) [inline modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/java/com/liferay/frontend/taglib/sample/web/internal/display/context/SearchPaginatorDisplayContext.java:54]: `SamplePortletKeys.CLAY_SAMPLE` is no longer valid here, we need to use a new constant as this portlet is different

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init.jsp [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+<%@ page import="java.util.List" %>
+
+<liferay-theme:defineObjects />
+
+<%@ include file="/init-ext.jsp" %>
+
+<%
+SampleDisplayContext sampleDisplayContext = (SampleDisplayContext)request.getAttribute(SamplePortletKeys.SAMPLE_DISPLAY_CONTEXT);
+SearchPaginatorDisplayContext searchPaginatorDisplayContext = (SearchPaginatorDisplayContext)request.getAttribute(SamplePortletKeys.SEARCH_PAGINATOR_DISPLAY_CONTEXT);
+%>
```
Comment (dsanz) [inline modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init.jsp:27]: In this PR we just need these 2 display context objects, the rest must be removed as not needed

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/.lfrbuild-portal
[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/.lfrbuild-portal:1]: This may have been product decisions, it makes sense to include some sample widgets and fragments in DXP. But by default, this should not be the case.

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/constants/ClaySamplePortletKeys.java (hunk 4 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/constants/ClaySamplePortletKeys.java]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/display/context/ClaySampleDisplayContext.java (hunk 5 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/display/context/ClaySampleDisplayContext.java]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/display/context/SearchPaginatorDisplayContext.java (hunk 6 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/display/context/SearchPaginatorDisplayContext.java]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/portlet/ClaySamplePortlet.java (hunk 7 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/portlet/ClaySamplePortlet.java]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/init.jsp (hunk 8 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/init.jsp]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/java/com/liferay/frontend/taglib/sample/web/internal/display/context/SearchPaginatorDisplayContext.java (hunk 9 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/java/com/liferay/frontend/taglib/sample/web/internal/display/context/SearchPaginatorDisplayContext.java]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init.jsp (hunk 10 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init.jsp:27]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/.lfrbuild-portal
[the remark this answers is not in the document]
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/.lfrbuild-portal:1]: I've removed the flag file from current PR, we'll see later if we want to remove it from some other module.

File: modules/test/playwright/tests/frontend-taglib/frontend-taglib-sample-web/liferay-ui/searchPaginator.spec.ts
Comment (chestofwonders) [reply modules/test/playwright/tests/frontend-taglib/frontend-taglib-sample-web/liferay-ui/searchPaginator.spec.ts:1]: It was just a suggestion for organizing the tags. If we agree with it and the naming of the folder, we can copy it to Sample Web, but looking at it later, it might be better to simply delete this subfolder, since we only have one at the moment. What do you think?

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/ClaySampleMultiselectPropsTransformer.js
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/ClaySampleMultiselectPropsTransformer.js:1]: I have removed all the js from the PR, so that file no longer exists

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/ClaySampleMultiselectPropsTransformer.js (hunk 3 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/ClaySampleMultiselectPropsTransformer.js]: js files can be removed, because they are not needed. but should we keep the js project structure? I mean `tsconfig, package.json`, etc. files for future js additions?

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/index.js
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/index.js:1]: Done, please see comment below

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/ClaySampleMultiselectPropsTransformer.js (hunk 3 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/js/ClaySampleMultiselectPropsTransformer.js]: I've removed the whole js structure, as is not required for this PR

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/view.jsp (hunk 2 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/view.jsp]: I've removed the navigation structure, since we have only one element

File: modules/test/playwright/tests/frontend-taglib/frontend-taglib-sample-web/liferay-ui/searchPaginator.spec.ts
Comment (chestofwonders) [reply modules/test/playwright/tests/frontend-taglib/frontend-taglib-sample-web/liferay-ui/searchPaginator.spec.ts:1]: I've better removed the whole subfolder, since it creates more confusion. When elements number growth later we should agree for a folder structure that fits better

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/search_paginator.jsp
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/search_paginator.jsp:1]: All code in `clay-sample-web` is not needed now, right? As we have the same sample in `sample-web`

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/bnd.bnd [hunk 11]
```diff
@@ -0,0 +1,5 @@
+Bundle-Name: Liferay Frontend Taglib Sample Web
+Bundle-SymbolicName: com.liferay.frontend.taglib.sample.web
+Bundle-Version: 1.0.0
+Export-Package: com.liferay.frontend.taglib.sample.web.constants
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-sample-web/bnd.bnd:4]: Why do we need to export constants package? A sample module should normally not export anything.

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/derived-config.txt [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+com/liferay/frontend/taglib/sample/web/internal/display/context/SearchPaginatorDisplayContext.<init>(Lcom/liferay/portal/kernel/util/Portal;Ljavax/portlet/RenderRequest;Ljavax/portlet/RenderResponse;)V
+com/liferay/frontend/taglib/sample/web/internal/display/context/SearchPaginatorDisplayContext.getSearchContainer()Lcom/liferay/portal/kernel/dao/search/SearchContainer;:UNKNOWN
+org/apache/jsp/partials/search_005fpaginator_jsp.<clinit>()V
+org/apache/jsp/partials/search_005fpaginator_jsp.<init>()V
+org/apache/jsp/partials/search_005fpaginator_jsp.getDependants()Ljava/util/List;:SAFE
+org/apache/jsp/partials/search_005fpaginator_jsp._jspService(Ljavax/servlet/http/HttpServletRequest;Ljavax/servlet/http/HttpServletResponse;)V^0:0,UNKNOWN^1:1,UNKNOWN
+com/liferay/frontend/taglib/sample/web/constants/SamplePortletKeys.<init>()V
+com/liferay/frontend/taglib/sample/web/internal/portlet/SamplePortlet.<init>()V
+com/liferay/frontend/taglib/sample/web/internal/portlet/SamplePortlet.doDispatch(Ljavax/portlet/RenderRequest;Ljavax/portlet/RenderResponse;)V^0:0,UNKNOWN^1:1,UNKNOWN
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-sample-web/derived-config.txt:10]: These look like some auto-generated junk artifacts, I don't know where they are coming from. This should not be committed.

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init-ext.jsp [hunk 13]
```diff
@@ -0,0 +1,6 @@
+<%--
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+--%>
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init-ext.jsp:6]: We don't need to commit this file, there is nothing in it.

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init.jsp [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+<%@ page import="com.liferay.frontend.taglib.sample.web.constants.SamplePortletKeys" %><%@
+page import="com.liferay.frontend.taglib.sample.web.internal.display.context.SearchPaginatorDisplayContext" %>
+
+<liferay-theme:defineObjects />
+
+<%@ include file="/init-ext.jsp" %>
+
+<%
+SearchPaginatorDisplayContext searchPaginatorDisplayContext = (SearchPaginatorDisplayContext)request.getAttribute(SamplePortletKeys.SEARCH_PAGINATOR_DISPLAY_CONTEXT);
+%>
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init.jsp:22]: This should not be in `init.jsp`, a common file for many tag JSPs to come. Its scope is the search paginator tag.

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/view.jsp [hunk 15]
```diff
@@ -0,0 +1,10 @@
+<%--
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<liferay-util:include page="/partials/search_paginator.jsp" servletContext="<%= application %>" />
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/view.jsp:10]: Fair enough, we can implement navigation with the second tag. Looks a bit silly on it's own, but I suppose it's fine for now.

File: modules/test/playwright/yarn.lock
Comment (markocikos) [inline modules/test/playwright/yarn.lock:1]: This should not be committed. There is no need for you to run `yarn install` on playwright root.

**Review by markocikos — CHANGES_REQUESTED** [review]
@chestofwonders Please see comments inline

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init-ext.jsp (hunk 13 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init-ext.jsp:6]: I've removed it. There are plenty of similar ones around, couldn't it be that they have some internal function?

Comment (markocikos) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init-ext.jsp:6]: I'm quite sure they have no internal function, they are just leftover junk.

File: modules/test/playwright/yarn.lock
Comment (chestofwonders) [reply modules/test/playwright/yarn.lock:1]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/view.jsp (hunk 15 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/view.jsp:10]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init.jsp (hunk 14 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/init.jsp]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/bnd.bnd (hunk 11 above)
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/bnd.bnd]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/search_paginator.jsp
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/search_paginator.jsp:1]: Done!

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/java/com/liferay/frontend/taglib/sample/web/constants/SamplePortletKeys.java [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+
+package com.liferay.frontend.taglib.sample.web.constants;
+
+/**
+ * @author Miguel Arroyo
+ */
+public class SamplePortletKeys {
+
+	public static final String SAMPLE_PORTLET =
+		"com_liferay_sample_web_portlet_SamplePortlet";
```
[the remark this answers is not in the document]
Comment (chestofwonders) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/java/com/liferay/frontend/taglib/sample/web/constants/SamplePortletKeys.java:14]: Let's give it a try...cross fingers

File: modules/apps/change-tracking/change-tracking-web/npmscripts.config.js [hunk 17]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  "build": {
+    "babel": false,
+    "bundler": false,
+    "exports": false,
+    "main": false,
+    "sass": false,
+    "tsc": false
+  }
+};
```
Comment (markocikos) [inline modules/apps/change-tracking/change-tracking-web/npmscripts.config.js:10]: Why is these two `npmscripts.config.js` files now committed? They look unrelated :thinking:

**Review by markocikos — COMMENTED** [review]
@chestofwonders Just one question inline

File: modules/apps/change-tracking/change-tracking-web/npmscripts.config.js (hunk 17 above)
Comment (chestofwonders) [reply modules/apps/change-tracking/change-tracking-web/npmscripts.config.js]: Mmm it seems to be created by the toolset... I've checked for more garbage files and delete them.

**Review by markocikos — APPROVED** [review]
@chestofwonders We can try pushing it like this, but there are two non-blocking awkward things about this this PR:
1. `SamplePortlet` is just so generic. I'd call this `TaglibSamplePortlet`.
2.  Including just a single JSP under `partials`. I'd implement `<clay:navigation-bar>` here with just one item.
